### ISSUE TITLE
[airgap rke2] custom rke2 cluster SUC system-agent-upgrader containers stuck in re-deploying/terminating state `didn't match pod anti-affinity rules.`

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#35819]
**Rancher Server Setup**
- Rancher version: v2.6.3-rc4
- Installation option (Docker install/Helm Chart): helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):rke1
- Proxy/Cert Details: valid letsencrypt certs

**Information about the Cluster**
- Kubernetes version: 1.21.7+rke2r1
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):custom

**Describe the bug**

* custom cluster (single node all roles) SUC deployment is continuously deploying and terminating containers with image `rancher/system-agent:v0.1.1-rc5-suc`

**To Reproduce**
 
* deploy an airgap HA setup 
* deploy a downstream rke2 cluster

**Result**
* pods/containers are continuously restarting for system-agent-upgrader

**Expected Result**
Expecting there to be no restarts/reinstalling pods or workloads. 

<img width="242" alt="Screen Shot 2021-12-08 at 5 27 06 PM" src="https://user-images.githubusercontent.com/16691014/145330567-4ba31c4f-383b-4cc1-9c5b-06a23aac088d.png">

**Comment by dweomer** [comment]
@slickwarren some questions:
- does this only fail for RKE1 local cluster?
- does it matter which version of rke2 the imported cluster is running?
- does this same scenario also fail when not air-gapped?

**Comment by thedadams** [comment]
@dweomer Not able to reproduce this in a non-airgapped scenario in Digital Ocean.
1. Deployed an RKE1 cluster (single-node)
2. Installed cert-manager and rancher (v2.6.3-rc4)
3. Created a custom cluster in Rancher (RKE2 v1.21.7-rke2r2)
4. Ran the registration command on the other Digital Ocean droplet.
5. Cluster comes up active. I see the `system-upgrade-controller` deployment installed on the downstream cluster briefly, but it is removed rather quickly.

**Comment by dweomer** [comment]
log snippet shared with me by @thedadams ~ 12/10: [log.txt](https://github.com/rancher/rancher/files/7705226/Untitled.txt)

i noticed that the plan is constantly changing, flapping between two states. attached are the decoded values from the `objectset.rio.cattle.io/applied` annotation on the job(s) representing applied state: 
- [1.txt](https://github.com/rancher/rancher/files/7705248/1.txt)
- [2.txt](https://github.com/rancher/rancher/files/7705249/2.txt) (did see some noise on stderr about decoding some info. think it was from `base64 -d` but possibly it was `jq` in the pipeline complaining)

the diff for these is:
```diff
--- 1.json	2021-12-09 14:13:57.707992369 -0700
+++ 2.json	2021-12-09 14:13:52.827980985 -0700
@@ -7,6 +7,7 @@
       "objectset.rio.cattle.io/owner-namespace": "cattle-system",
       "upgrade.cattle.io/ttl-seconds-after-finished": "900"
     },
+    "creationTimestamp": null,
     "labels": {
       "objectset.rio.cattle.io/hash": "5578292ae46c6979afa82eb77388959b0ddb938a",
       "plan.upgrade.cattle.io/system-agent-upgrader": "8a0b20eef98cf5f73ebdb2e10aff4508e1c86b2290af1feb6248a85f",
```
which represents a changing value for `.spec.metadata.creationTimestamp` which shouldn't cause the flapping.

**Comment by slickwarren** [comment]
@dweomer getting back to your questions:
* This was tested in airgap with an rke1 local cluster
* 1.22 was not working for rke2 with this RC, we will ask for a new rancher RC today to verify this
* This is not breaking for a non-airgap setup with the same version(s)

**Comment by dweomer** [comment]
https://github.com/rancher/system-upgrade-controller/releases/tag/v0.8.1

**Comment by thedadams** [comment]
@dweomer Thanks. I tested that version and it works for the setup I tested.

**Comment by Jono-SUSE-Rancher** [comment]
@cmurphy Can you update SUC images in system-charts and charts to v0.8.1?

**Comment by thedadams** [comment]
### Root cause
The root cause here was that SUC was continually deleting and recreating the upgrade job.

### What was fixed, or what changes have occurred
Default values are added to some fields, so when `Apply` was used with these default values missing, Wrangler thought that the job was changing and needed to be deleted and recreated.

### Areas or cases that should be tested
I was only able to reproduce this issue in air-gapped environment. So, the thing that should be tested here is the scenario listed in the description.

### What areas could experience regressions?
There should be no regressions as long as the test case in the description passes.

### Are the repro steps accurate/minimal?
Yes.

**Comment by brendarearden** [comment]
Available to test in v2.6.3-rc7

**Comment by slickwarren** [comment]
blocked from fully validating by https://github.com/rancher/rancher/issues/35485#issuecomment-997071182

**Comment by brendarearden** [comment]
Available to test in rancher/rancher v2.6.3-rc8

**Comment by slickwarren** [comment]
tested on v2.6.3-rc8 in an HA Airgap setup:
* deploy rke2 downstream cluster (1.21.7)
  * SUC image is deployed and working properly -- pass
* upgrade rke2 downstream cluster -- pass
